net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml

```diff
@@ -15,6 +15,8 @@
                 </name>
                 <pubkey type="TextField">
                     <Required>N</Required>
+                    <mask>/^[a-zA-Z0-9+\/]{43}=$/</mask>
```

Comment:
This isn't all proper base64. AFAIK the "=" can be zero to three times for padding. The 43 is probably a guess as well?

@AdSchellevis what do you think about a Base64Field type that validates this correctly?

Reply:
@fichtner we can easily add a field type for it, which would be way better than spreading TextField's all over the place.

Reply:
I like :)

Reply:
it someone opens an issue in core, I'll promise to look into it.

Reply:
https://github.com/opnsense/core/issues/4398 thanks!

Reply:
43 + = isn't a guess: I basically copy paste the check perform for the psk field, ensuring the regex actually match well a key generated by wg pubkey. I like the idea of a Base64Field but I'm not familiar enough with opnsense src to have an idea of what is better. If Base64Field allow different size, it should be added something to check the string lengh

Reply:
If this is to be done for this field, I suggest you also update the psk field format to Base64Field

Reply:
@mimugmail https://github.com/opnsense/core/commit/43bd9812b681ae055201c28768052a436c6bb308 there you go.